vortex-serde/src/file/read/builder/mod.rs

```diff
@@ -165,6 +172,11 @@ impl<R: VortexReadAt> VortexReadBuilder<R> {
             })
             .transpose()?;
 
+        // Default: fallback to single-threaded tokio dispatcher.
+        let io_dispatcher = self
+            .io_dispatcher
+            .unwrap_or_else(|| IoDispatcher::new_tokio(1));
```

Comment:
WDYT about first trying to use the `current` runtime?

Reply:
I had considered that, it is a bit tricky b/c you can't launch !Send futures on current runtime so need to duplicate some of the TokioDispatcher logic, but perhaps is less surprising for consumers

Reply:
The downside is that given we do CPU-intensive operations on the current runtime, it's probably net-better to offload the IO operations even when you're using tokio. I'm open to suggestions about what you think is the right approach here

Reply:
API-wise I think this decision makes it very easy for the user to get really subpar performance. 
The thing I like best here would be to make the dispatcher a required argument in the builder's constructor, that way the user HAS to make a choice, and we can provide some sane defaults that are explicitly outlined.
Another way is to just have a higher thread count, but I'm not sure there's a good way to pick a default number that works for everyone.

Reply:
We will if we `tokio::spawn` (or other runtime equivalent functionality) each range so they can make progress without being polled directly, but I'm not sure how that would interact with the single threaded runtimes.

Reply:
we talked - I now believe it should ok, but we probably want some benchmarks at some point to see how the number of runtime threads impacts performance for different runtimes.